frontend/src/components/AddConcepts.tsx

```diff
@@ -0,0 +1,134 @@
+import DataTable, {createTheme} from "react-data-table-component";
+
+import {useDataGetter, DataGetter} from "../state/DataGetter";
+
+interface Concept {
+  readonly concept_id: number;
+  readonly concept_name: string;
+  readonly domain_id: string;
+  readonly vocabulary_id: string;
+  readonly concept_class_id: string;
+  readonly standard_concept: string;
+  readonly concept_code: string;
+  readonly invalid_reason: string;
+  readonly domain_cnt: number;
+  readonly domain: string;
+  readonly total_cnt: number;
+  readonly distinct_person_cnt: string;
+}
+
+const columns = [
+  {
+    name: "Concept Name",
+    selector: (row) => row.concept_name,
+    sortable: true,
+    width: 200,
+  },
+  {
+    selector: row => row.domain_id,
+    name: "Domain",
+    sortable: true,
+    width: 200,
+  },
+  {
+    selector: row => row.total_cnt,
+    name: "Total Count",
+    sortable: true,
+    width: 100,
+  },
+  {
+    selector: row => row.distinct_person_cnt,
+    name: "Distinct Person Count",
+    sortable: true,
+    width: 100,
+  },
+];
+
+export function AddConcepts() {
+  return <ConceptStringSearch />;
+
+}
+// from https://stackoverflow.com/a/66167322/1368860
+function ConceptStringSearch() {
+  const dataGetter: DataGetter = useDataGetter();
+  let concepts: Concept[] = [];
+  const [searchText, setSearchText] = React.useState("");
+  const [concept_ids, setConceptIds] = React.useState(concepts);
+  const lastRequest = React.useRef(null);
+
+  // this effect will be fired every time searchText changes
+  React.useEffect(() => {
+    (async () => {
+      // setting min lenght for searchText
+      if (searchText.length >= 3) {
+        // updating the ref variable with the current searchText
+        lastRequest.current = searchText;
+        const r = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.concept_search, searchText)
+        if (lastRequest.current === searchText) {
+          console.log("response is valid!", r);
+          setConceptIds(r);
+        } else {
+          console.log("discarding api response", searchText, lastRequest.current);
+        }
+      }
+    })();
+  }, [searchText]);
+
+  return (
+    <div className="App" style={{paddingLeft: 100}}>
+      <h1>Concept Search</h1>
+      <input style={{width: 300}} type="text" placeholder="match characters in concept name"
+             onChange={(e) => setSearchText(e.target.value)}
+             value={searchText}
+             autoFocus={true}
+      />{'\u00A0'}{'\u00A0'}{'\u00A0'}{concept_ids.length ? concept_ids.length.toLocaleString() + ' concept_ids' : ""}
+      <ConceptTable concept_ids={concept_ids} />
+    </div>
+  );
+}
+
+function ConceptTable(props) {
+  let {concept_ids} = props;
+  const dataGetter = useDataGetter();
+  const [concepts, setConcepts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const totalRows = concept_ids.length;
+  const [perPage, setPerPage] = useState(10);
```

Comment:
```suggestion
  const [perPage, setPerPage] = useState(30);
```
Maybe set it higher?
Though I guess while we're developing it, doing it in plain orders of magnitude 10, 100, etc, is easier to reason about.

Reply:
I guess it looks like there's a way for the user to change how many rows there are as well?